### DISCUSSION TITLE
How could I use the js function in the formula?

### DISCUSSION THREAD

**Discussion by hiocean** [RafaelGB/obsidian-db-folder#449] [Q&A] [answered]
For example, I put the js file myfile.js in a folder called myfolder. and the in the js file, i write a function, e.g. like:

```js
function func1(x) {
    return x
}
module.exports = func1;
```

then, how should i use such func in the formula?
like 
${db.myfile.js.func1(0)} 
${db.js.func1(0)} 
is not working.

**Comment by RafaelGB** [comment, marked as answer]
Moved to Q&A disccussions

You need to select where the js files are saved 

<img width="293" alt="config table" src="https://user-images.githubusercontent.com/11924043/193227965-4cb2c7aa-f158-4f2d-94c1-a76028076be4.PNG">

The use of them could be consulted [here](https://rafaelgb.github.io/obsidian-db-folder/features/Formulas/)
It just explain the basics.

The name of the js file should be the same of your exposed function:
**func1.js**
```js
function func1(x) {
    return x
}
module.exports = func1;
```

And the use was correct:
```js
${db.js.func1(0)}
```

**Reply by hiocean** [reply]
Thanks!

**Reply by DannyHatcher-Git** [reply]
I am trying to understand the formula property myself. 

The JS you have written there. 
From what I understand you are making a function called funct1 and returning (outputting) x.

I don't understand what the module.export = funct1; does?
Is there any specific output for this or is it placeholder information for something larger?

**Reply by RafaelGB** [reply]
to group all the user formulas the plugin is using the `eval` functionality of javascript

after load an entire js file, the plugin save internally that module.export as new function just like Templater. In fact its Templater plugin code adapted to Dbfolder  needs

**Reply by DannyHatcher-Git** [reply]
So the module.export = (function) makes the function a file that can be called inside the formula property in the database?

What does this function example do apart from make a function with x?
I am trying to get my head around the JS file and the interaction with the database information, like the metadata etc

**Comment by vagrant-psyche** [comment]
I know the reason. If you activate the JS in plugin global settings, then you open the old database that was created previously, you will find that you still don't activate the JS in database settings.

So the solution are
1.  activate the JS in database settings which is in the old database created previously.
2.  create a new database, you will find JS is activated  the JS in database settings by default.

@RafaelGB 
@hiocean
